daemon/cmd/daemon.go

```diff
@@ -356,6 +356,7 @@ func NewDaemon(ctx context.Context, cancel context.CancelFunc, epMgr *endpointma
 	if option.Config.ReadCNIConfiguration != "" {
 		netConf, err = cnitypes.ReadNetConf(option.Config.ReadCNIConfiguration)
 		if err != nil {
+			log.WithError(err).Error("Unable to read CNI configuration")
 			return nil, nil, fmt.Errorf("unable to read CNI configuration: %w", err)
```

Comment:
Here though it seems like we're potentially duplicating the error log + wrap the error which the caller will also log. Is this because the error is being returned from `NewDaemon()`?

Reply:
Yup, 

The discussion we initially had was when we *just* return an error, we loose context of the subsystem it came from. So by adding the logs back, we now get the subsystem information back.

Reply:
I see. Well as long as `unable to read CNI configuration: _____` is unique, I think we can figure out how to debug an issue with purely the wrapped error here, regardless of the subsystem. My biggest concern would be particularly any sort of generic error (syscall errors are common in this regard) that don't get wrapped somewhere up the stack, then get logged at a really high level.

Reply:
So should we back out the log statements?

Reply:
I personally like the subsystem headings/prefixes and would be sad to see them go. It definitely makes debugging go faster. If we're just doing this in the `cmd` package it seems like an acceptable tradeoff to have a few duplicated error messages.

If we start doing this everywhere then I think we should consider one or the other.

Reply:
I'm OK with this. We'll log the message twice, but one of those log messages will be a stacked here, where the logged sring contains multiple error strings concat'd potentially.